simplecpp.h

```diff
@@ -76,18 +76,15 @@ namespace simplecpp {
      */
     class SIMPLECPP_LIB Location {
     public:
-        explicit Location(const std::vector<std::string> &f) : files(f) {}
+        Location() = default;
+        Location(unsigned int fileIndex, unsigned int line, unsigned int col)
+            : fileIndex(fileIndex)
+            , line(line)
+            , col(col)
+        {}
 
         Location(const Location &loc) = default;
-
-        Location &operator=(const Location &other) {
-            if (this != &other) {
-                fileIndex = other.fileIndex;
-                line = other.line;
-                col  = other.col;
-            }
-            return *this;
-        }
+        Location &operator=(const Location &other) = default;
```

Comment:
This was lacking the assignment of `files`. I have no idea how that could have ever been a valid object - beside the fact that containing a reference is even problematic to begin with.